### ISSUE TITLE
Crash when opening detail view for non https website.

### ISSUE THREAD

**Issue by alex1704** [mitmproxy/mitmproxy#4387]
#### Problem Description
Application crashes when open Detail tab of website with non secure connection.

#### Steps to reproduce the behavior:
1. open in browser http://example.com/
2. go to flow
3. open Detail tab

```
Traceback (most recent call last):
  File "/Users/alex/work/code/python/mitmproxy/mitmproxy/master.py", line 56, in run_loop
    loop()
  File "/Users/alex/work/code/python/mitmproxy/venv/lib/python3.9/site-packages/urwid/main_loop.py", line 287, in run
    self._run()
  File "/Users/alex/work/code/python/mitmproxy/venv/lib/python3.9/site-packages/urwid/main_loop.py", line 385, in _run
    self.event_loop.run()
  File "/Users/alex/work/code/python/mitmproxy/venv/lib/python3.9/site-packages/urwid/main_loop.py", line 1494, in run
    reraise(*exc_info)
  File "/Users/alex/work/code/python/mitmproxy/venv/lib/python3.9/site-packages/urwid/compat.py", line 58, in reraise
    raise value
  File "/usr/local/Cellar/python@3.9/3.9.0_1/Frameworks/Python.framework/Versions/3.9/lib/python3.9/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/Users/alex/work/code/python/mitmproxy/venv/lib/python3.9/site-packages/urwid/raw_display.py", line 416, in <lambda>
    wrapper = lambda: self.parse_input(
  File "/Users/alex/work/code/python/mitmproxy/venv/lib/python3.9/site-packages/urwid/raw_display.py", line 515, in parse_input
    callback(processed, processed_codes)
  File "/Users/alex/work/code/python/mitmproxy/venv/lib/python3.9/site-packages/urwid/main_loop.py", line 412, in _update
    self.process_input(keys)
  File "/Users/alex/work/code/python/mitmproxy/venv/lib/python3.9/site-packages/urwid/main_loop.py", line 513, in process_input
    k = self._topmost_widget.keypress(self.screen_size, k)
  File "/Users/alex/work/code/python/mitmproxy/mitmproxy/tools/console/window.py", line 316, in keypress
    k = super().keypress(size, k)
  File "/Users/alex/work/code/python/mitmproxy/venv/lib/python3.9/site-packages/urwid/container.py", line 1135, in keypress
    return self.body.keypress( (maxcol, remaining), key )
  File "/Users/alex/work/code/python/mitmproxy/mitmproxy/tools/console/window.py", line 50, in keypress
    ret = super().keypress(size, key)
  File "/Users/alex/work/code/python/mitmproxy/venv/lib/python3.9/site-packages/urwid/container.py", line 1135, in keypress
    return self.body.keypress( (maxcol, remaining), key )
  File "/Users/alex/work/code/python/mitmproxy/venv/lib/python3.9/site-packages/urwid/container.py", line 1135, in keypress
    return self.body.keypress( (maxcol, remaining), key )
  File "/Users/alex/work/code/python/mitmproxy/mitmproxy/tools/console/tabs.py", line 41, in keypress
    self.change_tab((self.tab_offset + 1) % n)
  File "/Users/alex/work/code/python/mitmproxy/mitmproxy/tools/console/tabs.py", line 34, in change_tab
    self.show()
  File "/Users/alex/work/code/python/mitmproxy/mitmproxy/tools/console/tabs.py", line 73, in show
    body = self.tabs[self.tab_offset % len(self.tabs)][1](),
  File "/Users/alex/work/code/python/mitmproxy/mitmproxy/tools/console/flowview.py", line 176, in view_details
    return flowdetailview.flowdetails(self.view, self.flow)
  File "/Users/alex/work/code/python/mitmproxy/mitmproxy/tools/console/flowdetailview.py", line 58, in flowdetails
    c = sc.certificate_list[0]
IndexError: tuple index out of range

mitmproxy has crashed!
```

#### System Information
Mitmproxy: 7.0.0.dev (+281, commit 4909198)
Python:    3.9.0
OpenSSL:   OpenSSL 1.1.1i  8 Dec 2020
Platform:  macOS-10.15.7-x86_64-i386-64bit

**Comment by mhils** [comment]
Fixed by ff624d0. Thanks for testing mitmproxy from master! 😃

**Comment by alex1704** [comment]
Thanks for awesome tool!
